### PR TITLE
feat(plugin owner): adds logic

### PR DESCRIPTION
**Pull request by yuliroth**
This PR adds plugin owner logic as described here:  https://applicaster.atlassian.net/wiki/spaces/PRON/pages/604701295/TF-CMS+-+Plugin+Account

### How should this be manually tested?

1. bundle install
2. in accounts, Go to applications -> zapp -> add "Per Account Roles" `plugin_developer`
3. still in accounts -> go to your user, and add in Permissions -> zapp `plugin_developer`
4. then try to run zappifest locally with the following variations:

```
ruby lib/zappifest.rb publish --manifest  <path-to-manifest-json-file>  --access-token <zapp-access-token>  --account <account-id-that-you-have-permission> --base-url=http://localhost:4000/api/v1/admin --accounts-url=http://accounts.test/api/v1 --new

ruby lib/zappifest.rb publish --manifest <path-to-manifest-json-file>  --access-token <zapp-access-token>  --account <account-id-that-you-have-permission> --base-url http://localhost:4000/api/v1/admin --accounts-url=http://accounts.test/api/v1

ruby lib/zappifest.rb publish --manifest  <path-to-manifest-json-file>  --access-token <zapp-access-token>  --account <account-id-that-you-DONT-have-permission> --base-url http://localhost:4000/api/v1/admin --accounts-url=http://accounts.test/api/v1 --new

ruby lib/zappifest.rb publish --manifest  <path-to-manifest-json-file>  --access-token <zapp-access-token>  --account <account-id-that-you-DONT-have-permission> --base-url http://localhost:4000/api/v1/admin --accounts-url=http://accounts.test/api/v1

ruby lib/zappifest.rb  get_account_plugins --access-token <zapp-access-token> --account <account-id-that-you-have-permission>
```

### CODE REVIEW

**Comment by snooky23** [comment]
@yuliroth - what about documentation in our developer portal? :)

**Comment by yuliroth** [comment]
@snooky23 documentation will be there

File: README.md [hunk 1]
```diff
@@ -7,58 +7,70 @@
...
 
 #### Prerequisites
+
 Reach Applicaster support team to generate User access-token.
 
 The tool allow you to publish the plugin to Zapp.
 
-Run `zappifest publish --manifest <path-to-manifest-json-file> --access-token <zapp-access-token>`
+Run `zappifest publish --manifest <path-to-manifest-json-file> --access-token <zapp-access-token> --account <account-id>`
 If you want to publish a manifest for a new plugin, with a new identifier, you need to use the `--new` option.
```
Comment (simonvaucher) [inline README.md]: 👍

File: lib/plugin_base.rb [hunk 2]
```diff
@@ -43,4 +47,69 @@ def get_markdown_data(markdown_file)
...
+  end
+
+  def request_params
+    {}.tap do |params|
+      params["id"] = @existing_plugin["id"] unless @existing_plugin.nil?
+      params["access_token"] = @access_token
+      params["plugin[name]"] = @name
+      params["plugin[category]"] = @manifest["type"]
+      params["plugin[external_identifier]"] = @identifier
+      params["plugin[whitelisted_account_ids][]"] = @manifest["whitelisted_account_ids"] || @plugin_account
```
Comment (simonvaucher) [inline lib/plugin_base.rb]: @plugin_account is  a single string - I think it should be converted to an array, something like  `[@plugin_account]`

File: lib/zappifest.rb [hunk 3]
```diff
@@ -87,16 +90,28 @@
   c.action do |args, options|
     options.default access_token: ENV["ZAPP_TOKEN"]
     options.default manifest: args.first
-
+    
     VersionHelper.new(c).check_version
     NetworkHelpers.validate_token(options.access_token) unless options.override_url
+    current_user = NetworkHelpers.current_user(options.access_token).body
```
Comment (simonvaucher) [inline lib/zappifest.rb]: Now I know why it fails:
1. Before we had `NetworkHelpers.validate_token(options.access_token) unless options.override_url`
and `current_user` was _never_ called locally because of the `unless`
2. Now when we get to `current_user` it fails since it tries to access
`#{ACCOUNTS_URL}/api/v1/users/current.json` 
3. ACCOUNTS_URL = "https://accounts.applicaster.com/api/v1" 😢 
4. We should change the ACCOUNTS_URL to use `accounts.test` locally and then `current_user` should be okay

Bottom line:
- Local publishing NEVER used the local zapp token in `accounts.test`. I think it should (in a totally different PR)

Comment (yuliroth) [reply lib/zappifest.rb]: @simonvaucher add at your local envs `ACCOUNTS_URL` to accounts.test

Comment (simonvaucher) [reply lib/zappifest.rb]: How will that help? Line 6 in NetworkHelpers is:
`ACCOUNTS_URL = "https://accounts.applicaster.com/api/v1"`

Comment (simonvaucher) [reply lib/zappifest.rb]: We can always _fix_ this line to this:
```
ACCOUNTS_URL = ENV["ACCOUNTS_URL"] || ""https://accounts.applicaster.com/api/v1"
```
+ modify the README

Comment (simonvaucher) [reply lib/zappifest.rb]: Fixed in the meanwhile with #162

File: README.md [hunk 4]
```diff
@@ -7,64 +7,78 @@
...
 ### Publish
 
 #### Prerequisites
+
 Reach Applicaster support team to generate User access-token.
 
 The tool allow you to publish the plugin to Zapp.
 
-Run `zappifest publish --manifest <path-to-manifest-json-file> --access-token <zapp-access-token>`
+Run `zappifest publish --manifest <path-to-manifest-json-file> --access-token <zapp-access-token> --account <account-id>`
```
Comment (ranmeirman) [inline README.md]: We can actually remove the --manifest flag, you can just run publish <path_to_manifest>

File: README.md [hunk 5]
```diff
@@ -7,64 +7,78 @@
...
 Note: It is recommended to set the access-token as environment variable `ZAPP_TOKEN`.
 By doing so, `access_token` param for publishing will not be required.
 ```
 
 #### Updating existing plugin
+
 Check the plugin id on [Zapp](https://zapp.applicaster.com/admin/plugins) (under the relevant plugin versions).
 
-Run `zappifest publish --plugin-id <plugin-id> --manifest <path-to-manifest-json-file> --access-token <zapp-access-token>`
+Run `zappifest publish --plugin-id <plugin-id> --manifest <path-to-manifest-json-file> --access-token <zapp-access-token> --account <account-id>`
```
Comment (ranmeirman) [inline README.md]: same comment as above

File: lib/account_helper.rb [hunk 6]
```diff
@@ -0,0 +1,22 @@
+class AccountHelper
+    include NetworkHelpers
+  
+    def initialize(user, account)
+      @user = user
+      @account = account
+    end
+  
+    def valid_account?(options)
+      existing_account = NetworkHelpers.get_accounts_list(options).body.find { |account| account["old_id"] == @account }
```
Comment (ranmeirman) [inline lib/account_helper.rb]: this will throw and error if the body is not an array and an error happend

File: lib/account_helper.rb [hunk 7]
```diff
@@ -0,0 +1,22 @@
+class AccountHelper
+    include NetworkHelpers
+  
+    def initialize(user, account)
```
Comment (ranmeirman) [inline lib/account_helper.rb]: where is `new` being called?

Comment (yuliroth) [reply lib/account_helper.rb]: @ranmeirman at zappifest.rb line 100

**Comment by yuliroth** [comment]
@ranmeirman 
1. `zappifest get_account_plugins --account <account-id>` will be in separate account
2. i didn't refactor NetworkHelpers..
3. init is not working in master as well, now with @simonvaucher will try to understand why

**Comment by simonvaucher** [comment]
> it fails after the first input of the `zappifest init` command. NetworkHelpers is marked as module but you added initialize method which is in use to create new class.
fix in => #163 😬

File: README.md (hunk 4 above)
Comment (yuliroth) [reply README.md]: @ranmeirman i prefer to do it in separate PR, not related to this PR
